doc/tutorials/dnn/dnn_text_spotting/dnn_text_spotting.markdown

```diff
@@ -0,0 +1,366 @@
+sha: b215b5fb85d1c606d1f56f03e8c841025e72733d
+parameter setting: -inputHeight=736, -inputWidth=1280;
+description: This model is trained on ICDAR2015, so it is only able to detect English text instances.
+
+- DB_TD500_resnet50.onnx:
+url: https://drive.google.com/uc?export=dowload&id=1_9KB6QkIghp2VNZti_dJ3Araz-nDZBsL
+sha: 6d3b7b84fc23906be7d583a8cedef727cc351fad
+parameter setting: -inputHeight=736, -inputWidth=736;
+description: This model is trained on MSRA-TD500, so it is able to detect English and Chinese text instances.
+
+- DB_TD500_resnet18.onnx:
+url: https://drive.google.com/uc?export=dowload&id=1f0dfqG_-Fiq4ywdrRhq8JH9PXCcJDO01
+sha: dd19abe7e2326ac660ce2733be1cb7032f76d458
+parameter setting: -inputHeight=736, -inputWidth=736;
+description: This model is trained on MSRA-TD500, so it is able to detect English and Chinese text instances.
+
+```
+
+The models with Res50 have a better accuracy, while those with Res18 have a faster inference speed.
+
+## Images for Testing
+```
+Text Recognition:
+url: https://drive.google.com/uc?export=dowload&id=1nMcEy68zDNpIlqAn6xCk_kYcUTIeSOtN
+sha: 89205612ce8dd2251effa16609342b69bff67ca3
+
+Text Detection:
+url: https://drive.google.com/uc?export=dowload&id=149tAhIcvfCYeyufRoZ9tmc2mZDKE_XrF
+sha: ced3c03fb7f8d9608169a913acf7e7b93e07109b
+```
+
+## Example for Text Recognition
+
+```cpp
+#include <iostream>
+#include <fstream>
+
+#include <opencv2/core.hpp>
+#include <opencv2/highgui/highgui.hpp>
+#include <opencv2/dnn.hpp>
+
+using namespace cv;
+using namespace dnn;
+
+int main()
+{
+    // Load model weights
+    TextRecognitionModel model("path/to/crnn_cs.onnx");
+
+    // Load vocabulary
+    // vocabulary should be changed according to the text recognition model
+    std::ifstream vocFile;
+    vocFile.open("path/to/alphabet_94.txt");
+    CV_Assert(vocFile.is_open());
+    String vocLine;
+    std::vector<String> vocabulary;
+    while (std::getline(vocFile, vocLine)) {
+        vocabulary.push_back(vocLine);
+    }
+
+    // Normalization parameters
+    double scale = 1.0 / 127.5;
+    Scalar mean = Scalar(127.5);
+
+    // The input shape is fixed to (width=100, height=32)
+    Size inputSize = Size(100, 32);
+
+    model.setInputParams(scale, inputSize, mean);
+
+    // more decoding method will be supported in future
+    String decodeType = "CTC-greedy";
+
+    // Load a cropped text line image
+    // you can find cropped images for testing in "Images for Testing"
+    int rgb = 1; // This should be changed according to the model input requirement.
+    Mat image = imread("path/to/text_rec_test.png", rgb);
+
+    // Inference
+    std::vector<String> recResults;
+    model.recognize(image, decodeType, vocabulary, recResults);
+    std::cout << recResults[0] << std::endl;
+
+    return 0;
+}
+
+```
+Input image:
+
+![Picture example](https://github.com/HannibalAPE/images/blob/master/opencv/text_rec_test.png)
```

Comment:
External links on images are not allowed.
Make image as a part of documentation tree instead (but please avoid adding images with bad license terms)